TreeTracker/UI/Views/Messaging/ChatList/ChatListViewModel.swift

```diff
@@ -0,0 +1,182 @@
+
+                } else {
+                    let newChat = Chat(
+                        title: message.subject ?? "Admin",
+                        type: messageType,
+                        messages: [message]
+                    )
+                    chatList.insert(newChat, at: 0)
+                }
+
+            default:
+                let newChat = Chat(
+                    title: message.subject ?? "",
+                    type: messageType,
+                    messages: [message]
+                )
+                chatList.append(newChat)
+            }
+        }
+
+        self.chatList = chatList
+    }
+
+    func updateUnreadMessagesCount(indexPath: IndexPath) {
+        let messagesToUpdate = chatList[indexPath.row].messages
+        messagingService.updateUnreadMessages(messages: messagesToUpdate)
+    }
+
+}
+
+// MARK: - Navigation
+extension ChatListViewModel {
+
+    func chatSelected(indexPath: IndexPath) {
+        let selectedChat = chatList[indexPath.row]
+        updateUnreadMessagesCount(indexPath: indexPath)
+
+        switch selectedChat.type {
+        case .message:
+            coordinatorDelegate?.chatListViewModel(self, didSelectMessages: planter)
+        case .announce:
+            coordinatorDelegate?.chatListViewModel(self, didSelectAnnounce: selectedChat)
+        case .survey, .surveyResponse:
+
+            guard let message = selectedChat.messages.first,
+                  let questions = message.survey?.questions?.array as? [SurveyQuestion] else {
+                return
+            }
+
+            let survey = SurveyViewModel.Survey(
+                surveyId: message.survey?.uuid ?? "",
+                title: message.survey?.title ?? "",
+                questions: questions.map({ question in
+                    return SurveyViewModel.Question(
+                        prompt: question.prompt ?? "",
+                        choices: question.choices ?? [])
+                }),
+                surveyResponse: message.surveyResponse ?? [],
+                response: message.survey?.response ?? false
+            )
+
+            coordinatorDelegate?.chatListViewModel(self, didSelectSurvey: survey, planter: planter)
+        }
+    }
+
+}
+
+// MARK: - Nested Types
+extension ChatListViewModel {
+
+    struct Chat {
+        let title: String
+        let type: MessageType
+        var messages: [MessageEntity]
+
+        var image: UIImage {
+            switch type {
+            case .message:
+                return Asset.Assets.forest.image
+            case .announce:
+                return Asset.Assets.gardening.image
+            case .survey:
+                return Asset.Assets.logo.image
+            case .surveyResponse:
+                return Asset.Assets.people.image
+            }
+        }
+
+        var unreadCount: Int {
+            messages.reduce(0) { $0 + ($1.unread ? 1 : 0) }
+        }
+    }
+
+    enum MessageType: String, Decodable {
```

Comment:
We seem to be duplicating models from the core sdk here. Could we try utilise those instead?

Reply:
Ok I'll make the change